internal/app/networkd/pkg/networkd/netconf.go

```diff
@@ -43,11 +44,31 @@ func buildOptions(device machine.Device, hostname string) (name string, opts []n
 		}
 
 		opts = append(opts, nic.WithAddressing(s))
+	case device.DHCP == false:
+		//Allow master interface without any addressing if VLANs exist
+		if device.CIDR == "" && len(device.Vlans) > 0 {
```

Comment:
looks like this check is not needed, as `device.CIDR == ""` is processed in the `case` above

Reply:
True, it came there by mistake during debugging why DHCP was still coming after network reset before I realized nic.go have a force DHCP if addressing methods missing.